airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/destination_process/DockerizedDestination.kt

```diff
@@ -106,20 +123,89 @@ class DockerizedDestination(
 
         logger.info { "Executing command: ${cmd.joinToString(" ")}" }
         process = ProcessBuilder(cmd).start()
+        // Annoyingly, the process's stdin is called "outputStream"
+        destinationStdin = BufferedWriter(OutputStreamWriter(process.outputStream))
+    }
+
+    override fun run() {
+        runBlocking {
+            withContext(Dispatchers.IO) {
+                launch {
+                    // Consume stdout. These should all be properly-formatted messages.
+                    // Annoyingly, the process's stdout is called "inputStream".
+                    val destinationStdout = Scanner(process.inputStream)
+                    while (destinationStdout.hasNextLine()) {
```

Comment:
How will this work with the test pattern of write some, read some, write some more?

Reply:
Or does the "input" stream scanner just block until the process halts?

Reply:
yep, exactly. if the process is still running but has nothing new on stdout, this just blocks.